go/adbc/driver/databricks/stmt_record_reader.go

```diff
@@ -130,7 +135,7 @@ func newRecordReader(
 		chunkChan:       make(chan chunkResponse),
 		activeChunk:     nil,
 
-		schema: nil,
+		schema: arrowSchema,
```

Comment:
A bit of a change of plans from what we've talked:

Add a new field here called `derivedSchema` and keep `schema` as `nil` from the start. Now see my comments in the `Schema()` method.